commit 27d13af71c3af6b2f9b60556d2c046dbb6e36e23
Author: Mike Frysinger <vapier@gentoo.org>
Date:   Mon Mar 14 17:44:49 2016 -0400

    include sys/sysmacros.h in more places
    
    Since glibc is moving away from implicitly including sys/sysmacros.h
    all the time via sys/types.h, include the header directly in more
    places.  This seems to cover most makedev/major/minor usage.

diff --git a/src/shared/macro.h b/src/shared/macro.h
index c34441d..b36a956 100644
--- a/src/shared/macro.h
+++ b/src/shared/macro.h
@@ -23,6 +23,7 @@
 #include <inttypes.h>
 #include <stdbool.h>
 #include <sys/param.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 
 #define _printf_(a,b) __attribute__ ((format (printf, a, b)))
diff --git a/src/shared/util.h b/src/shared/util.h
--- a/src/shared/util.h
+++ b/src/shared/util.h
@@ -36,6 +36,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <dirent.h>
 #include <stddef.h>
 #include <unistd.h>
diff --git a/src/systemd/sd-device.h b/src/systemd/sd-device.h
--- a/src/systemd/sd-device.h
+++ b/src/systemd/sd-device.h
@@ -22,6 +22,7 @@
 ***/
 
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <stdint.h>
 
 #include "_sd-common.h"
